src/main/java/mate/academy/spring/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.spring.security;
+
+import java.util.Optional;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(username);
+        if (userOptional.isEmpty()) {
+            throw new UsernameNotFoundException("User not found");
+        }
+        User user = userOptional.orElseThrow();
+        org.springframework.security.core.userdetails.User.UserBuilder builder
+                = org.springframework.security.core.userdetails.User.withUsername(username);
```

Comment:
Добав import org.springframework.security.core.userdetails.User.UserBuilder;
тоді можна буде трохи скоротити
```suggestion
        User user = userOptional.orElseThrow();
        UserBuilder builder = org.springframework.security.core.userdetails.User.withUsername(username);
```